spectrum.js

```diff
@@ -136,7 +136,9 @@
             var current = p[i];
             if(current) {
                 var tiny = tinycolor(current);
-                var c = tiny.toHsl().l < 0.5 ? "sp-thumb-el sp-thumb-dark" : "sp-thumb-el sp-thumb-light";
+                var hsl = tiny.toHsl();
+                var darkColor = hsl.l + (1 - hsl.a);
+                var c = darkColor < 1.0 ? "sp-thumb-el sp-thumb-dark" : "sp-thumb-el sp-thumb-light";
```

Comment:
You are right. I obviously tested only the corner cases that I were interested in.

Regarding your proposal: I checked the code in mostReadable and it has the same problem that it does not consider the alpha values. I would agree on using it, but in the case of low alpha values it would still just decide on the whole color. I think there is no point in using a more complex function, if it doesn't solve the actual problem.

I will reconsider the formula to fit every need.